app/src/main/java/ch/epfl/sdp/peakar/gallery/GalleryActivity.java

```diff
@@ -66,4 +59,11 @@ private void setupGallery(){
         recyclerView.setLayoutManager(new GridLayoutManager(this, COLUMNS));
         recyclerView.setAdapter(galleryAdapter);
     }
+
+    @Override
+    protected void onRestart() {
+        super.onRestart();
+        setupGallery();
+        findViewById(R.id.gallery_recyclerview).setOnTouchListener(new OnSwipeTouchListener(this));
```

Comment:
I see that this is the only activity to set the listener again at onRestart. Is there a reason for why this need it but the others don't?

Reply:
No, this was added for testing purposes and forgot to remove it.